Assets/_BForBoss/_Core/Scripts/Boss/DerekHealthBehaviour.cs

```diff
@@ -1,70 +1,87 @@
 
+        [SerializeField] private DerekHealthView _healthView;
+
         private int _elapsedHealth;
         private float _threshold;
         private DerekContextManager.Vulnerability _vulnerabilityState = DerekContextManager.Vulnerability.Invulnerable;
-        
+
+        private bool _canReceiveDamage = false;
+
         private event Action<DerekHealthViewState> OnStateChanged;
         private event Action OnThresholdHit;
 
-        public DerekContextManager.Vulnerability VulnerabilityState
+        public bool CanReceiveDamage
         {
-            get => _vulnerabilityState;
+            get => _canReceiveDamage;
             set
             {
-                _vulnerabilityState = value;
-                OnStateChanged?.Invoke(MapToState());
-            } 
+                _canReceiveDamage = value;
+                _healthView.SetState(new DerekHealthViewState(HealthPercentage, !_canReceiveDamage));
+            }
         }
 
-        public void Initialize(
-            Action onThresholdHit, 
-            Action<DerekHealthViewState> onStateChanged)
+        private float HealthPercentage => (float)_elapsedHealth / _maxHealth;
+
+        public void Initialize(Action onThresholdHit)
         {
             OnThresholdHit = onThresholdHit;
-            OnStateChanged = onStateChanged;
         }
-        
-        private void Awake()
+
+        public void Reset()
         {
             _elapsedHealth = _maxHealth;
+            CanReceiveDamage = false;
         }
 
-        private float HealthPercentage => _elapsedHealth / _maxHealth;
+        private void Awake()
+        {
+            this.PanicIfNullObject(_healthView, nameof(_healthView));
+
+            if (_maxHealth <= 0)
+            {
+                PanicHelper.Panic(new Exception($"{nameof(_maxHealth)} needs to be a positive number, instead it is set to {_maxHealth}"));
+            }
+
+            _elapsedHealth = _maxHealth;
+        }
 
         public void OnCollided(Vector3 collisionPoint, Vector3 collisionNormal)
         {
-            if (VulnerabilityState == DerekContextManager.Vulnerability.Invulnerable)
+            if (!CanReceiveDamage)
             {
                 return;
             }
+
             _elapsedHealth -= _damagePerShot;
+            _healthView.SetState(new DerekHealthViewState(HealthPercentage, !CanReceiveDamage));
+            
             if (HealthPercentage <= _threshold)
             {
                 OnThresholdHit?.Invoke();
             }
-            OnStateChanged?.Invoke(MapToState());
         }
 
-        public void SetNextThreshold(float healthPercentage)
+        public void SetThreshold(float healthPercentage, Color healthBarColor)
         {
+            if (healthPercentage < 0.0f || healthPercentage > 1.0f)
+            {
+                Perigon.Utility.Logger.LogWarning("Percentage is not between 0 and 1, clamping", LoggerColor.Yellow, "derekboss");
```

Comment:
I feel like the clamping will happen enough times that it shouldn't be a warning. That it's a common usage.